PokeAppiOS/Views/PokemonDetail/Cells/EvolutionVariantTableViewCell.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+
+using Foundation;
+using UIKit;
+using SharedCode.Model.Api;
+using System.Collections.Generic;
+using CoreAudioKit;
+using SharedCode.Controller;
+using SharedCode.Services;
+using SharedCode.Model.DB;
+using SharedCode.Util;
+
+namespace PokeAppiOS.Views.PokemonDetail.Cells
+{
+	public partial class EvolutionVariantTableViewCell : UITableViewCell
+    {
+		public static readonly NSString Key = new NSString ("EvolutionVariantTableViewCell");
+		public static readonly UINib Nib;
+
+        public UIColor PrimaryColor;
+
+        IPokemonDetailController controller = IocContainer.GetDependency<IPokemonDetailController>();
+
+        static EvolutionVariantTableViewCell ()
+		{
+			Nib = UINib.FromName ("EvolutionVariantTableViewCell", NSBundle.MainBundle);
+		}
+
+		protected EvolutionVariantTableViewCell (IntPtr handle) : base (handle)
+		{
+			// Note: this .ctor should not contain any initialization logic.
+		}
+
+		public List<ResultItem> EvolutionChainPairList
+		{
+			set
+			{
+                fromLabel.Text = StringUtils.ToTitleCase(value[0].Name);
+                toLabel.Text = StringUtils.ToTitleCase(value[1].Name);
+                loadImages(value);
+                if (PrimaryColor != null)
+                {
+                    fromLabel.TextColor = PrimaryColor;
+                    toLabel.TextColor = PrimaryColor;
+                    arrowImageView.TintColor = PrimaryColor;
+                }
+            }
+		}
+
+        private void loadImages(List<ResultItem> evolutionChainPairList)
+        {
+            loadEvolutionFromImage(evolutionChainPairList[0]);
```

Comment:
Maybe you can add a validation in case there is no data like FirstOrDefault or check if is not null

Reply:
FirstOrDefault and LastOrDefault added.

Reply:
Maybe you forgot to commit that part

Reply:
Commited.